src/main/java/com/uber/cadence/samples/hello/HelloSaga.java

```diff
@@ -0,0 +1,128 @@
+import com.uber.cadence.client.WorkflowClient;
+import com.uber.cadence.client.WorkflowOptions;
+import com.uber.cadence.worker.Worker;
+import com.uber.cadence.workflow.*;
+
+import java.time.Duration;
+
+import static com.uber.cadence.samples.common.SampleConstants.DOMAIN;
+
+/**
+ * Demonstrates implementing saga transaction and compensation logic using Cadence.
+ */
+public class HelloSaga {
+  static final String TASK_LIST = "HelloSaga";
+
+  public interface ChildWorkflowOperation {
+    @WorkflowMethod
+    void execute(int amount);
+  }
+
+  public static class ChildWorkflowOperationImpl implements ChildWorkflowOperation {
+    ActivityOperation activity = Workflow.newActivityStub(ActivityOperation.class);
+
+    public void execute(int amount) {
+      activity.execute(amount);
+    }
+  }
+
+  public interface ChildWorkflowCompensation {
+    @WorkflowMethod
+    void compensate(int amount);
+  }
+
+  public static class ChildWorkflowCompensationImpl implements ChildWorkflowCompensation {
+    ActivityOperation activity = Workflow.newActivityStub(ActivityOperation.class);
+
+    public void compensate(int amount) {
+      activity.compensate(amount);
+    }
+  }
+
+  public interface ActivityOperation {
+    @ActivityMethod(scheduleToCloseTimeoutSeconds = 2)
+    void execute(int amount);
+
+    @ActivityMethod(scheduleToCloseTimeoutSeconds = 2)
+    void compensate(int amount);
+  }
+
+  public static class ActivityOperationImpl implements ActivityOperation {
+
+    public void execute(int amount) {
+      System.out.println("ActivityOperationImpl.execute() is called with amount " + amount);
+    }
+
+    public void compensate(int amount) {
+      System.out.println("ActivityCompensationImpl.compensate() is called with amount " + amount);
+    }
+  }
+
+  public interface SagaWorkflow {
+    /**
+     * Main saga workflow.
+     * Here we execute activity operation twice (first from a child workflow, second directly using
+     * activity stub), add three compensation functions, and then throws some exception in workflow code.
+     * When we catch the exception, saga.compensate will run the compensation functions according
+     * to the policy specified in SagaOptions.
+     */
+    @WorkflowMethod
+    void execute();
+  }
+
+  public static class SagaWorkflowImpl implements SagaWorkflow {
+    ActivityOperation activity = Workflow.newActivityStub(ActivityOperation.class);
+
+    @Override
+    public void execute() {
+      Saga saga = new Saga(new Saga.Options.Builder().setParallelCompensation(false).build());
+      try {
+        ChildWorkflowOperation op1 = Workflow.newChildWorkflowStub(ChildWorkflowOperation.class);
+        op1.execute(10);
+        ChildWorkflowCompensation c1 = Workflow.newChildWorkflowStub(ChildWorkflowCompensation.class);
+        saga.addCompensation(c1::compensate, -10);
+
+        Promise<Void> result = Async.procedure(activity::execute, 20);
+        result.get();
```

Comment:
switch line 89 and 90 and add comment that this demonstrates how to compensate an async activity invocation.